task_sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -111,11 +115,109 @@ def get_template_context(self):
                 "ts_nodash_with_tz": ts_nodash_with_tz,
             }
             context.update(context_from_server)
+        # TODO: We should use/move TypeDict from airflow.utils.context.Context
         return context
 
-    def xcom_pull(self, *args, **kwargs): ...
+    def xcom_pull(
+        self,
+        task_ids: str | None = None,  # TODO: Simplify to a single task_id? (breaking change)
+        dag_id: str | None = None,
+        key: str = "return_value",  # TODO: Make this a constant (``XCOM_RETURN_KEY``)
+        include_prior_dates: bool = False,  # TODO: Add support for this
+        *,
+        map_indexes: int | None = None,
```

Comment:
Name says plural, but type only allows one value

Reply:
Resolved in https://github.com/apache/airflow/pull/45075/commits/ae097e8a3cf26c2d3f9df6f97895a29a07a08500